# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/SocketProcessBridgeChild.h
# Commit: c414d82a5325
# Full Hash: c414d82a532508523d40ce63111cb30b2a1076be
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-02-22 05:26:22
# Regressor Bug: 1521879
# File Overlap Count: 1
# Description:
#   Bug 1521879 - Part 1: IPC-based MediaTransport implementation r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D17273
# ==============================================================================

diff -r 3a182b05e2c1 -r c414d82a5325 netwerk/ipc/SocketProcessBridgeChild.h
--- a/netwerk/ipc/SocketProcessBridgeChild.h	Thu Feb 21 19:01:32 2019 +0200
+++ b/netwerk/ipc/SocketProcessBridgeChild.h	Thu Feb 21 16:42:12 2019 +0000
@@ -22,10 +22,10 @@
   NS_DECL_ISUPPORTS
   NS_DECL_NSIOBSERVER
 
-  static bool Create(Endpoint<PSocketProcessBridgeChild>&& aEndpoint);
   static already_AddRefed<SocketProcessBridgeChild> GetSingleton();
-  static void EnsureSocketProcessBridge(std::function<void()>&& aOnSuccess,
-                                        std::function<void()>&& aOnFailure);
+  typedef MozPromise<RefPtr<SocketProcessBridgeChild>, nsCString, false>
+      GetPromise;
+  static RefPtr<GetPromise> GetSocketProcessBridge();
 
   mozilla::ipc::IPCResult RecvTest();
   void ActorDestroy(ActorDestroyReason aWhy) override;
@@ -34,8 +34,12 @@
   bool Inited() const { return mInited; };
   ProcessId SocketProcessPid() const { return mSocketProcessPid; };
 
+  dom::PMediaTransportChild* AllocPMediaTransportChild() override;
+  bool DeallocPMediaTransportChild(dom::PMediaTransportChild* aActor) override;
+
  private:
   DISALLOW_COPY_AND_ASSIGN(SocketProcessBridgeChild);
+  static bool Create(Endpoint<PSocketProcessBridgeChild>&& aEndpoint);
   explicit SocketProcessBridgeChild(
       Endpoint<PSocketProcessBridgeChild>&& aEndpoint);
   virtual ~SocketProcessBridgeChild();